homeassistant/bootstrap.py

```diff
@@ -474,9 +475,7 @@ def _get_domains(hass: core.HomeAssistant, config: dict[str, Any]) -> set[str]:
     """Get domains of components to set up."""
     # Filter out the repeating and common config section [homeassistant]
     domains = {
-        domain
-        for key in config
-        if (domain := conf_util.domain_from_config_key(key)) != core.DOMAIN
+        domain for key in config if (domain := cv.domain_key(key)) != core.DOMAIN
```

Comment:
This is done before the config validation of the config besides the core config, right? Should we suppress `vol.Invalid` here? Does this mean the user won't even know that a domain is invalid?

Reply:
Perfect!